### PR TITLE
*: adapt needs for operator support

### PR DESCRIPTION
**Pull request by xhebox**
Signed-off-by: xhe <xw897002528@gmail.com>

### What problem does this PR solve?

Issue Number: ref #69 

Problem Summary: A big PR including changes needed to adapt tidb-operator

What is changed and how it works:

1. github actions now detects version by `go.mod`
2. split parts from `pkg` into `/lib`. Otherwise tidb-operator is not able to import `weirctl` due to the complex dependency conflicts.
3. Makefile and dockerfile improvement.
4. `server.Run` returns no error
5. Added `--pub_addr` for `weirproxy` to expose the correct advertise address in k8s, simplify `xxx-urls` config, too
6. Removed unnecessary labels in metrics, but currently there is no usefull metrics for `weirproxy`, see #69 
7. Removed `pingcap/log` in `util/security`

### Check List

Tests 

- [x] Unit test
- [x] Integration test
- [x] Manual test (add detailed scripts or steps below)
Tested with k8s
- [ ] No code

Notable changes

- [ ] Has configuration change
- [ ] Has HTTP API interfaces change (Don't forget to [add the declarative for API](https://github.com/tikv/pd/blob/master/docs/development.md#updating-api-documentation))
- [ ] Has weirctl change
- [ ] Other user behavior changes

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: lib/config/file.go [hunk 1]
```diff
@@ -42,13 +42,13 @@ func NewNamespacesFromDir(nsdir string) (*NamespaceDir, error) {
 		nspath: make(map[string]string),
 	}
 
-	yamlFiles, err := listAllYamlFiles(c.dir)
+	nFiles, err := listAllFiles(c.dir, ".toml")
```
Comment (djshow832) [inline lib/config/file.go:45]: Why change to toml files this time? Is it required by the operator?

File: lib/config/file.go [hunk 2]
```diff
@@ -57,13 +57,13 @@ func NewNamespacesFromDir(nsdir string) (*NamespaceDir, error) {
...
 		c.cfgs[cfg.Namespace] = &cfg
-		c.nspath[cfg.Namespace] = yamlFile
+		c.nspath[cfg.Namespace] = nfile
 	}
 
 	return c, nil
 }
 
-func listAllYamlFiles(dir string) ([]string, error) {
+func listAllFiles(dir, ext string) ([]string, error) {
```
Comment (djshow832) [inline lib/config/file.go:66]: Isn't it duplicate with
https://github.com/pingcap/TiProxy/blob/dc6452506016dbcdba179d7253555b14ae5e117d/lib/cli/namespace.go#L36

File: pkg/server/server.go [hunk 3]
```diff
@@ -91,10 +94,37 @@ func NewServer(ctx context.Context, cfg *config.Config, logger *zap.Logger) (srv
...
+			}
+			peerPort = strconv.Itoa(peerPortNum + 1)
+		}
+		peerAddr := *apiAddr
+		peerAddr.Host = fmt.Sprintf("%s:%s", peerAddr.Hostname(), peerPort)
+		etcd_cfg.LPUrls = []url.URL{peerAddr}
+		peerAddrAdvertise := *apiAddr
+		peerAddrAdvertise.Host = fmt.Sprintf("%s:%s", pubAddr, peerPort)
+		etcd_cfg.APUrls = []url.URL{peerAddrAdvertise}
+
```
Comment (djshow832) [inline pkg/server/server.go:127]: Extract the etcd part to a new function.

File: go.mod [hunk 4]
```diff
@@ -111,16 +110,21 @@ require (
...
+	golang.org/x/time v0.0.0-20220722155302-e5dcc9cfc0b9 // indirect
+	google.golang.org/genproto v0.0.0-20220822174746-9e6da59bd2fc // indirect
+	google.golang.org/protobuf v1.28.1 // indirect
 	gopkg.in/natefinch/lumberjack.v2 v2.0.0 // indirect
+	gopkg.in/yaml.v2 v2.4.0 // indirect
+	gopkg.in/yaml.v3 v3.0.1 // indirect
 	sigs.k8s.io/yaml v1.2.0 // indirect
 )
+
+replace github.com/pingcap/TiProxy/lib => ./lib
```
Comment (djshow832) [inline go.mod:130]: Why?

Comment (xhebox) [reply go.mod:130]: Same reason as `parser as sub module of tidb`. Can not test without pushing lib first.

File: lib/config/file.go (hunk 2 above)
Comment (xhebox) [reply lib/config/file.go]: Correct, I am removing `/lib/config/file`.

File: lib/config/file.go (hunk 1 above)
Comment (xhebox) [reply lib/config/file.go]: Kind of. Operator has built-in support for json and toml but not yaml.
Anyway, dynamic namespace will not be done by using configuration file. Also this is duplicated with `weirctl`.
It is not used at all, I removed this.

File: pkg/server/server.go (hunk 3 above)
Comment (xhebox) [reply pkg/server/server.go]: Done.

**Review by djshow832 — APPROVED** [review]

**Pull request merged**